metrics/sink_test.go

```diff
@@ -5,198 +5,188 @@ import (
 	"time"
 
 	"github.com/stretchr/testify/assert"
-	"github.com/stretchr/testify/require"
 )
 
-func TestCounterSink(t *testing.T) {
+func TestCounterSinkAddSample(t *testing.T) {
+	t.Parallel()
 	samples10 := []float64{1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 9.0, 100.0}
 	now := time.Now()
 
-	t.Run("add", func(t *testing.T) {
-		t.Run("one value", func(t *testing.T) {
-			sink := CounterSink{}
-			sink.Add(Sample{TimeSeries: TimeSeries{Metric: &Metric{}}, Value: 1.0, Time: now})
-			assert.Equal(t, 1.0, sink.Value)
-			assert.Equal(t, now, sink.First)
-		})
-		t.Run("values", func(t *testing.T) {
-			sink := CounterSink{}
-			for _, s := range samples10 {
-				sink.Add(Sample{TimeSeries: TimeSeries{Metric: &Metric{}}, Value: s, Time: now})
-			}
-			assert.Equal(t, 145.0, sink.Value)
-			assert.Equal(t, now, sink.First)
-		})
+	t.Run("one value", func(t *testing.T) {
+		t.Parallel()
+		sink := CounterSink{}
+		sink.Add(Sample{TimeSeries: TimeSeries{Metric: &Metric{}}, Value: 1.0, Time: now})
+		assert.Equal(t, 1.0, sink.Value)
+		assert.Equal(t, now, sink.First)
 	})
-	t.Run("format", func(t *testing.T) {
+	t.Run("values", func(t *testing.T) {
+		t.Parallel()
 		sink := CounterSink{}
 		for _, s := range samples10 {
 			sink.Add(Sample{TimeSeries: TimeSeries{Metric: &Metric{}}, Value: s, Time: now})
 		}
-		assert.Equal(t, map[string]float64{"count": 145, "rate": 145.0}, sink.Format(1*time.Second))
+		assert.Equal(t, 145.0, sink.Value)
+		assert.Equal(t, now, sink.First)
 	})
 }
 
-func TestGaugeSink(t *testing.T) {
+func TestCounterSinkRate(t *testing.T) {
```

Comment:
If you do think it's worth avoiding division by zero, as per my previous comment, could we add a test dedicated to that scenario here?